front/src/utils/duration.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable import/prefer-default-export */
+
+import dayjs from 'dayjs';
+
+export class Duration {
+  /** Number of milliseconds */
+  readonly ms: number;
+
+  constructor(ms: number) {
+    this.ms = ms;
+  }
+
+  static zero = new Duration(0);
+
+  /** Parse an ISO 8601 duration string. */
+  static parse(str: string) {
+    return new Duration(dayjs.duration(str).asMilliseconds());
+  }
+
+  /** Subtract two dates. */
+  static subtractDate(a: Date, b: Date) {
+    return new Duration(a.getTime() - b.getTime());
+  }
+
+  // Return the number of milliseconds, so that comparison operators work as expected.
+  valueOf() {
+    return this.ms;
+  }
+
+  /** Format this duration as an ISO 8601 string. */
+  toString() {
```

Comment:
Maybe toISOString would be a better name ?

Reply:
Indeed, and it would match `Date` better!

Reply:
Done